src/components/FormAddTodo/FormAddTodo.tsx

```diff
@@ -0,0 +1,121 @@
+import { useState } from 'react';
+
+import { TodoWithUser } from '../../types';
+
+import { findUserById } from '../../utils/findUserById';
+import { generateUniqueId } from '../../utils/generateUniqueId';
+
+import users from '../../api/users';
+import {
+  INVALID_TITLE_MESSAGE,
+  INVALID_USER_MESSAGE,
+} from './constants';
+import './FormAddTodo.scss';
+
+interface Props {
+  todos: TodoWithUser[];
+  onSubmit: (todo: TodoWithUser) => void;
+}
+
+export const FormAddTodo: React.FC<Props> = ({ onSubmit, todos }) => {
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [title, setTitle] = useState('');
+  const [isUserSelected, setIsUserSelected] = useState(true);
+  const [isTitleValid, setIsTitleValid] = useState(true);
+
+  function resetForm() {
+    setSelectedUserId(0);
+    setTitle('');
+  }
+
+  function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+
+    setIsTitleValid(() => Boolean(title));
+    setIsUserSelected(() => Boolean(selectedUserId));
```

Comment:
Not critical, but there is no much sense in using setter with function parameter if you don't use old state value